packages/grafana-ui/src/components/Actions/SuggestionsInput.tsx

```diff
@@ -0,0 +1,236 @@
+import { css } from '@emotion/css';
+import { autoUpdate, flip, shift, useFloating } from '@floating-ui/react';
+import { FormEvent, useCallback, useEffect, useRef, useState } from 'react';
+import * as React from 'react';
+
+import { GrafanaTheme2, VariableSuggestion } from '@grafana/data';
+
+import { useTheme2 } from '../../themes';
+import CustomScrollbar from '../CustomScrollbar/CustomScrollbar';
+import { DataLinkSuggestions } from '../DataLinks/DataLinkSuggestions';
+import { FieldValidationMessage } from '../Forms/FieldValidationMessage';
+import { Input } from '../Input/Input';
+import { Portal } from '../Portal/Portal';
+import { TextArea } from '../TextArea/TextArea';
+
+const modulo = (a: number, n: number) => a - n * Math.floor(a / n);
+const ERROR_TOOLTIP_OFFSET = 8;
+
+export enum HTMLElementType {
+  InputElement = 'input',
+  TextAreaElement = 'textarea',
+}
+
+interface SuggestionsInputProps {
+  value?: string | number;
+  onChange: (value: string, callback?: () => void) => void;
+  suggestions: VariableSuggestion[];
+  placeholder?: string;
+  invalid?: boolean;
+  error?: string;
+  width?: number;
+  type?: HTMLElementType;
+  style?: React.CSSProperties;
+}
+
+const getStyles = (theme: GrafanaTheme2, inputHeight: number) => {
+  return {
+    suggestionsWrapper: css({
+      boxShadow: theme.shadows.z2,
+    }),
+    errorTooltip: css({
+      position: 'absolute',
+      top: inputHeight + ERROR_TOOLTIP_OFFSET + 'px',
+      zIndex: theme.zIndex.tooltip,
+    }),
+    inputWrapper: css({
+      position: 'relative',
+    }),
+    // Wrapper with child selector needed.
+    // When classnames are applied to the same element as the wrapper, it causes the suggestions to stop working
+  };
+};
+
+export const SuggestionsInput = ({
```

Comment:
this whole file looks like mostly copy/paste of

https://github.com/grafana/grafana/blob/main/public/app/features/transformers/suggestionsInput/SuggestionsInput.tsx

can you explain what changes were done, and maybe we can reuse the other one with some additions?